src/drivers/indexeddb.js

```diff
@@ -947,36 +951,57 @@ function dropInstance(options, callback) {
     if (!options.name) {
         promise = Promise.reject('Invalid arguments');
     } else {
-        var dbPromise =
-            options.name === currentConfig.name && self._dbInfo.db
-                ? Promise.resolve(self._dbInfo.db)
-                : _getOriginalConnection(options);
+        const isCurrentDb =
+            options.name === currentConfig.name && self._dbInfo.db;
+
+        const dbPromise = isCurrentDb
+            ? Promise.resolve(self._dbInfo.db)
+            : _getOriginalConnection(options).then(function(db) {
+                  const dbContext = dbContexts[options.name];
+                  const forages = dbContext.forages;
+                  dbContext.db = db;
+                  for (var i = 0; i < forages.length; i++) {
```

Comment:
Seeing the `var` and also the regular `for` loop here doesn't seem very ES6-like either, can we do it a bit more functionally?